buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/VagrantTestPlugin.groovy

```diff
@@ -383,20 +383,43 @@ class VagrantTestPlugin implements Plugin<Project> {
         assert project.tasks.packagingTest != null
         Task packagingTest = project.tasks.packagingTest
 
+        assert project.tasks.clean != null
+        Task clean = project.tasks.clean
+
         /*
          * We always use the main project.rootDir as Vagrant's current working directory (VAGRANT_CWD)
          * so that boxes are not duplicated for every Gradle project that use this VagrantTestPlugin.
          */
+        File vagrantFile = new File(project.rootDir, 'Vagrantfile')
         def vagrantEnvVars = [
                 'VAGRANT_CWD'           : "${project.rootDir.absolutePath}",
-                'VAGRANT_VAGRANTFILE'   : 'Vagrantfile',
+                'VAGRANT_VAGRANTFILE'   : vagrantFile.name,
                 'VAGRANT_PROJECT_DIR'   : "${project.projectDir.absolutePath}"
         ]
 
+        File vagrantMarkersDir = new File(project.rootProject.buildDir, "vagrant")
+        clean.delete vagrantMarkersDir
```

Comment:
That's also what I thought but if I don't add this the marker file is not cleaned by `gradle clean`, maybe because it's the top level project but I'm not sure.